src/js/AccessibilityUtils.js

```diff
@@ -577,6 +577,40 @@ axs.utils.hasLabel = function(element) {
 };
 
 /**
+ * Determine if this element is disabled without considering the effect of a disabled ancestor.
+ * Disabled here means that the element should be considered disabled according to specification.
+ * This element may or may not be effectively disabled in practice as this is dependent on implementation.
+ * @param {Element} element An element to check.
+ * @return {boolean} True if the element is disabled.
+ */
+axs.utils.isElementDisabled = function(element) {
+    if(!element || element.nodeType !== Node.ELEMENT_NODE) {
+        return false;
+    }
+    var tagName = element.tagName.toUpperCase();
+    return (tagName in axs.constants.NATIVELY_DISABLEABLE && element.hasAttribute("disabled")) ||
+        element.getAttribute("aria-disabled") === "true";
+};
+
+/**
+ * Determine if this element is disabled directly or indirectly by a disabled ancestor.
+ * @see axs.utils.isElementDisabled
+ * @param {Element} element An element to check.
+ * @return {boolean} True if the element or one of its ancestors is disabled.
+ */
+axs.utils.isElementOrAncestorDisabled = function(element) {
+    if (axs.utils.isElementDisabled(element))
+        return true;
+    var parent = axs.utils.parentElement(element);
+    // fieldset disabled-ness does not flow to descendants of its first legend
+    if (parent && !axs.browserUtils.matchSelector(parent, "fieldset>legend:first-of-type")) {
+        return axs.utils.isElementOrAncestorDisabled(parent);
```

Comment:
I think this may need a little more subtlety - e.g. a `<span role=button>` inside a `<fieldset disabled>` _won't_ be disabled, right? You'd need to add `aria-disabled` somewhere in the ancestor chain.